File: integration-test/java/quarkus-app/src/main/java/com/qualityminds/lazyval/integration/boundary/persistence/jpa/PanacheRepository.java
package com.qualityminds.lazyval.integration.boundary.persistence.jpa;

import com.qualityminds.lazyval.integration.shared.Isbn;
import jakarta.enterprise.context.ApplicationScoped;

import java.util.List;
import java.util.UUID;

@ApplicationScoped
public class PanacheRepository implements io.quarkus.hibernate.orm.panache.PanacheRepository<JpaOrder> {

    JpaOrder findById(UUID id){
        return find("id", id).firstResult();
    }

    List<JpaOrder> findByIsbn(Isbn isbn){
        return find("isbn", isbn).list();
    }

}

File: integration-test/kotlin/quarkus-app/src/main/kotlin/com/qualityminds/lazyval/integration/boundary/persistence/jpa/PanacheRepository.kt
package com.qualityminds.lazyval.integration.boundary.persistence.jpa

import com.qualityminds.lazyval.integration.shared.Isbn
import io.quarkus.hibernate.orm.panache.kotlin.PanacheRepository
import jakarta.enterprise.context.ApplicationScoped
import java.util.UUID

@ApplicationScoped
class PanacheRepository : PanacheRepository<JpaOrder> {

    fun findById(id: UUID): JpaOrder? = find("id", id).firstResult()

    fun findByIsbn(isbn: Isbn): List<JpaOrder> = find("isbn", isbn).list()
}
